models/music_rate_in_proxy.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ *  music_rate_in_proxy.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "music_rate_in_proxy.h"
+
+#ifdef HAVE_MUSIC
+
+// Includes from sli:
+#include "arraydatum.h"
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+#include "integerdatum.h"
+
+// Includes from libnestutil:
+#include "compose.hpp"
+#include "logging.h"
+
+// Includes from nestkernel:
+#include "kernel_manager.h"
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::music_rate_in_proxy::Parameters_::Parameters_()
+  : port_name_( "rate_in" )
+  , channel_( 0 )
+{
+}
+
+nest::music_rate_in_proxy::Parameters_::Parameters_( const Parameters_& op )
+  : port_name_( op.port_name_ )
+  , channel_( op.channel_ )
+{
+}
+
+nest::music_rate_in_proxy::State_::State_()
+  : registered_( false )
+{
+}
+
+/* ----------------------------------------------------------------
+ * Parameter extraction and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::music_rate_in_proxy::Parameters_::get( DictionaryDatum& d ) const
+{
+  ( *d )[ names::port_name ] = port_name_;
+}
+
+void
+nest::music_rate_in_proxy::Parameters_::set( const DictionaryDatum& d, State_& s )
+{
+  // TODO: This is not possible, as P_ does not know about get_name()
```

Comment:
Is this still a TODO? Probably not the best idea to merge code with open TODOs in there :)

Reply:
hmm, this is a TODO copied over from `music_cont_in_proxy.cpp`, it's been there since we moved the code from svn to git. not sure how to resolve this?!

Reply:
Okay, if this is not a newly introduced TODO this is probably fine for now. However, it might be a good idea to open an issue (if there isn't already?) that raises the question if this is still a somehow important TODO oder should just be converted from TODO to an information.

Reply:
Thanks for digging in and creating the issue. I agree with @janhahne that this can stay in for now.